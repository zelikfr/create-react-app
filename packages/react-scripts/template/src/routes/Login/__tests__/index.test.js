import React from 'react'
import { mount } from 'enzyme'
import LoginPage from '../'

describe('LoginPage', () => {
  it('should create a component wrapped with LoginPage', () => {
    // When I render the LoginPage element
    const component = mount(<LoginPage />)
    // Then I should have the component to match the snapshot.
    expect(component).toMatchSnapshot()
  })
  it('should Login after click on button', () => {
    // When I render the LoginPage element
    const login = jest.fn()
    const component = mount(<LoginPage login={login} />)
    component.find('Button').simulate('click')
    // Then I should have the component to match the snapshot.
    expect(login).toHaveBeenCalled()
  })
})
