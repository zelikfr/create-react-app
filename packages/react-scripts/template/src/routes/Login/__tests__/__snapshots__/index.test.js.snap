// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should create a component wrapped with LoginPage 1`] = `
<Login>
  <div>
    <h1>
      Login
    </h1>
    <Button
      onClick={[Function]}
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Login to this website
      </button>
    </Button>
  </div>
</Login>
`;
